### PR TITLE
Make Hai space less profitable

### PR DESCRIPTION
**Pull request by Anarchist2**
It's been mentioned [here](https://github.com/endless-sky/endless-sky/pull/7251#issuecomment-1250264363) that, as an alterative to adding raid fleets to Hai space, we could instead reduce the delta of trade prices to make trading less profitable. This PR reduces trade profits in Hai space down to 40%, and also reduces the payment of many jobs in Hai space. In general, the reduced size of Hai space often means that there are more jobs going to a single destination, so the total payment per jump will frequently be higher. The largest payment nerfs were to the various Unfettered aid jobs, which used to pay in the realm of 1 million per job, for a 10-jump job that requires less than 100 cargo space. The payment on these jobs have been reduced to one-sixth, but the jobs now generate with larger cargo amounts to compensate.

### CODE REVIEW

**Review by Hurleveur — CHANGES_REQUESTED** [review]
Given this influences Hai I'm mentioning @MasterOfGrey (btw I'd appreciate being mentionned for modifications of missions I PRed)

It seems to me like the Hai jobs got a fine little nerf (I dont think we need to discourage ppl from doing jobs more but fine ig)

However, the Unfettered jobs got a huge hit, too big if you ask me. Yes they were very profitable, too much probably, which kinda was the point (some risks and very hard to stack them but they pay a lot, and they're not exactly common)
But one sixth is too much of a nerf, and increasing cargo doesnt help given that by the time you have this much cargo you're likely far ahead into the game already and woudnt use them, I think.
The idea is to helps adventurous captains to get started, not be a more retrained slightly better version of trading.

What we could do (and I'm planing to do) is reduce the money being given and give some kind of reputation boost that would allow being friendly to UHai without needing JDs. And I also want to add UHai raids to UHai space and Wah Ki + its nehighbors, increasing the risk of it all (which I think is better than lowering the reward)
This PR could just focus on nerfing trade which is the big balance issue.

I have no issue with making them pay up to half of what they do currently (this being a case of balance vs lore, they're desperate and should pay a lot as they do currently), but any more means they become normal job when they're only available trough high risk or after getting a JD (at which point the way to print money is Korath jobs in the EW, or the coalition that gives way more too)

**Comment by EjoThims** [comment]
Really not sure it's appropriate to bundle missions into trade reductions in general, but especially also such different sources and uses of missions, just because they're in the same general area. Especially when the nerfs are not even consistent values, but designed to reduce payments to a particular play style that doesn't even primarily benefit from these missions in this region, as even noted in that very same closed PR thread. https://github.com/endless-sky/endless-sky/pull/7251#issuecomment-1250187381

This should be split up into 3 different PRs to make sure each is appropriate and at appropriate levels. As they are, only the trade reduction one actually seems to be appropriate to me, especially since it's the only part actually addressing Hai space's use for super early trade fleet snowballing, and even then I'd prefer to see direct comparisons of outcome numbers showing it just being slowed, instead of made non-viable as an alternate strategy worth the risk of being able to get there to do it.

**Comment by Zitchas** [comment]
The reduction in commodity price spread seems quite appropriate. The Hai have an extremely stable society, don't have pirates, and have had millennia to work out their distribution networks. It's a low risk environment, and should be comparatively low profit margins for general distribution of commodities. (for that matter, the same applies to Coalition space, too)

Jobs and missions, though, I disagree with. They're already the worst way to make money (coming in a distant third behind trade fleet snowballing and ship looting/salvage/capturing), and the *only* means that a solo ship player has of making decent money that doesn't involve capturing. As such, the only real impact I see coming from reducing the payment for jobs and missions is making players feel like they are being pushed to either start trade fleet snowballing or start capturing ships for resale. This is a strong negative change, in my opinion. A *small* decrease might be appropriate, but a big decrease definitely is not.

In any case, everything other than the commodity prices should be stripped from this and handled separately.

**Comment by MasterOfGrey** [comment]
Just looking at the trade changes:
Certainly there's good reason to think trading in Hai space via *hyperdrive* should probably be less valuable.
However, there were some trade routes that made very good sense to be profitable with jump drives - and these should remain. After all, you are bypassing the standard logistics.
The value of these was that players who had invested in a moderate fleet of jump-capable vessels (who generally don't do trading cause they hate it), had a place where they could go collect some viable finances fairly quickly in non-optimised fleets by virtue of having jump drives - which your average death-snowballing trader won't have on all their ships.

Looking at some of the new single-jump trades you get:
- Heia Due to Io Lowe:
  - Plastic and heavy metals one way, medical the other, ~40cr/t profit
- Bote Asu to Ya Hai
  - Electronics, heavy metals, and plastics one way, luxury goods, equipment, and clothing the other, ~40cr/t profit

And to save space and time from me listing this ad nauseam: in general you're looking at *multiple goods*, with around ~40cr/t each way profit.

Now if we look at the jump routes where you skip at least one stop:
With these changes:
- Rati Cal -> Fah Soom : 
  - Equipment -> : 270 profit down to 108
  - Metal <-         : 120 profit down to 36
    - Skip 2 stops: baseline=120cr/t, Average 72cr/t | Outcome: non-viable (40% of previous profit would be average 132)

- Da Ent -> Dah Lest : 
  - Electronics -> : 230 profit down to 92
  - Food <-          : 110 profit down to 44
    - Skip 2 stops: baseline=120cr/t, Average 68cr/t | Outcome: non-viable (40% of previous profit would be average 114)

- Zuba Zub -> Bote Asu : 
  - Metal ->               : 170 profit down to 65
  - Heavy Metals ->  : 200 profit down to 80
  - Clothing <-          : 160 profit down to 64
  - Luxury Goods <-  : 190 profit down to 76
    - Skip 1 stop: baseline=80cr/t, Average 71.5cr/t | Outcome: sub-par (bang-on 40% but worse than regular trading)

- Zuba Zub -> Ula Mon : 
  - Metal ->               : 150 profit down to 60
  - Heavy Metals ->  : 200 profit down to 80
  - Clothing <-          : 160 profit down to 64
  - Luxury Goods <-  : 210 profit down to 84
    - Skip 1 stop: baseline=80cr/t, Average 72cr/t | Outcome: sub-par (bang-on 40% but worse than regular trading)

- Bote Asu -> Ula Mon (used to break things up): 
  - Medical -> : 90 profit down to 36
    - Outcome: sub-par

Conclusion:
While these changes have made a pretty neat market topography to be lore consistent for hyperdrive trading, it's almost completely killed the value of Hai space for players needing to do some lesser trading with jump drive ships to top up finances periodically.
This loss is something I think should be addressed before proceeding with the changes to trade prices.

**Comment by Anarchist2** [comment]
The problem with trying to balance both jump drive and hyperdrive trade routes in Hai space is that the extra jumps taken don't amount to a significant drop in profit. Take the Rati Cal/Fah Soom trade route: in this PR, doing one round trip in a Geocoris nets 296,790 credits in profit. If you have a jump drive, you have to pay 3,400 credits in salary, which amounts to 2.3% of your earnings. If you have a hyperdrive, you pay 13,600, or 4.6% of the profit. Overall, you only get 2.4% extra profit by switching from a jump drive to a hyperdrive. If you use the current trade prices, the extra profit margin gets even worse: more total payment means that crew salaries amount to even less of a percentage of your earnings.

The only ways to make using jump drives significantly better than having a hyperdrive within Hai space are to: a) reduce trade profits even more to make salaries take up a larger amount of the profits; b) increase crew salaries, for the same reason; or c) completely reroute Hai space so that jump drive shortcuts shave off even more time.

If you're willing to go beyond just Hai space, however, better shortcuts are available. Jumping to either Wanderer or Efreti space is impossible without a jump drive, and the Wah Oh/Alnitak route is shorted from 8 days to 2. If we're looking to reward traders with jump drives, doing it through these trades is a lot more feasible than trying to do something solely in Hai space.

**Review by Hurleveur — COMMENTED** [review]
I think its not just about profit it's also about time for the trade but then I'm not experienced with that, just commenting to clarify the requested changes of focusing on one thing was done

**Comment by MasterOfGrey** [comment]
> The problem with trying to balance both jump drive and hyperdrive trade routes in Hai space is that the extra jumps taken don't amount to a significant drop in profit. Take the Rati Cal/Fah Soom trade route: in this PR, doing one round trip in a Geocoris nets 296,790 credits in profit. If you have a jump drive, you have to pay 3,400 credits in salary, which amounts to 2.3% of your earnings. If you have a hyperdrive, you pay 13,600, or 4.6% of the profit. Overall, you only get 2.4% extra profit by switching from a jump drive to a hyperdrive. If you use the current trade prices, the extra profit margin gets even worse: more total payment means that crew salaries amount to even less of a percentage of your earnings.

You've kinda side-stepped the point I'm making though. If you're using a flotilla of trade ships that are crew-efficient, then you can just do trading, you don't need to go to the effort of acquiring jump drives.

The jump drives are of value to people who *don't* have optimised trade ships as part of their standard style of play. People who have maybe 2 armed freighters and a collection of heavy warships that they fly around with, or people who have a single utility vessel like a Bactrian that they might pause and temporarily store some outfits to trade for a minute, and then go back to doing other game things.
In other words: people who are **not** equipped to trade efficiently.

People who are using a large number of trade ships to do trading, would spend/waste more time acquiring the jump drives needed to make it work than they'd spend just normally doing the actual trading to meet their goal - the jump routes aren't for the regular traders; the jump routes are for *everyone else.*

> The only ways to make using jump drives significantly better than having a hyperdrive within Hai space are to: a) reduce trade profits even more to make salaries take up a larger amount of the profits; b) increase crew salaries, for the same reason; or c) completely reroute Hai space so that jump drive shortcuts shave off even more time.

Notably the routes I mentioned all either start, finish, or both, in dead ends, or could only be made efficient using hyperdrive by transiting Wah Ki (in the case of Zuba Zub - Bote Asu).

These are not efficient places to visit on regular trading routes, especially if you're also in the business of picking up jobs to leverage extra bonus cash along the way.
They're allowed to be a little more efficient (even with hyperdrive) because they have either hazards or opportunity costs associated with them.

> If you're willing to go beyond just Hai space, however, better shortcuts are available. Jumping to either Wanderer or Efreti space is impossible without a jump drive, and the Wah Oh/Alnitak route is shorted from 8 days to 2. If we're looking to reward traders with jump drives, doing it through these trades is a lot more feasible than trying to do something solely in Hai space.

And sure, that's a reasonable area of improvement to work on, but it's separate and more complicated than the point at hand.

**Comment by Anarchist2** [comment]
> People who are using a large number of trade ships to do trading, would spend/waste more time acquiring the jump drives needed to make it work than they'd spend just normally doing the actual trading to meet their goal - the jump routes aren't for the regular traders; the jump routes are for everyone else.

Then I don't understand what you want. Inherently, any trade route that would be appealing to someone doing casual trading on the side would be cracked open by people looking to optimise. What I'm getting at is that there isn't any meaningful difference between a 1-jump jump route or a 4-jump hyperdrive route.

Unless you're talking about making the jump routes more profitable, but not the most profitable routes in Hai space? Is that what you're saying?

**Comment by MasterOfGrey** [comment]
> Inherently, any trade route that would be appealing to someone doing casual trading on the side would be cracked open by people looking to optimise. 

"People who are using [...] trade ships [...] would waste more time acquiring the jump drives [...] [than just trading] to meet their goal".

Please tell me what part of that is unclear in the context of wanting the same cr/jump with hyperdrives to allow for *some* jump routes to stack those into a single profit for primary use by people who are not mainly traders.

**Comment by Zitchas** [comment]
> Inherently, any trade route that would be appealing to someone doing casual trading on the side would be cracked open by people looking to optimise.

It is more important that a solo ship player be able to make enough profit to be worth bothering to do than it is to nerf people flying fleets.

Beyond that, it feels like the discussion is getting side tracked here. The initial PR was talking about making flying in a very safe environment a bit less profitable. It seems there's some agreement that this is a good thing to do, we just need to work out the details. But now it is getting side tracked into talk about the profit margins of huge trading fleets and snowballing. 

In short: This isn't the place for that discussion. Trade fleet snowballing is a game-wide problem. The price gradient of Hai space isn't going to fix it, short of zeroing out the commodity gradient entirely. Solving that problem needs a solution that scales to it, which involves game mechanics, not just commodity price tweaks. 

Back on topic: If I'm understanding correctly, MasterofGrey wants there to be some decent traderoutes for Jump Drive equipped ships that can at least equal the profitability of hyperdrive routes. Trade routes that are useful for people flying solo or non-mercantile fleets.

**Comment by Hurleveur** [comment]
We can fix big fleet by making the price fluctuate faster, its really easy so no need to worry about that here.
But maybe big fleets balance was part of the worry here and the prices got a bit too low for a normal trader without JD because of that? When we think about it, human space isnt that unsafe except for pirate worlds (which dont have that good trading? I think they should, and the Unfettered do qualify as pirates in this context)
Cant a one ship with JD just use jobs, actually?
(But then we're removing one option here)

Also Wah Ki definitely isnt safe

**Comment by Anarchist2** [comment]
What I'm trying to say is that there isn't any difference between a "jump drive" route and a "hyperdrive" route within a single region.

> Cant a one ship with JD just use jobs, actually?

That's what I'm going for, yes.

**Comment by MasterOfGrey** [comment]
> What I'm trying to say is that there isn't any difference between a "jump drive" route and a "hyperdrive" route within a single region.

And what I'm saying, is that you would be wrong.

**Review by MasterOfGrey — APPROVED** [review]
Looks like my concerns have been addressed and the changes have been made with nuance to the surrounding systems.
Looks good. :)

**Review by Hurleveur — APPROVED** [review]
My concerns have been addresed too, but idk much about trade tbh so tke this with a grain of salt

**Comment by Hurleveur** [comment]
Also that'd be for another PR but I would like to add raids near Wah Ki (surrounding systems) since we now have more flexibility with that
I guess we'll up the price there a bit then

**Comment by ThrawnCA** [comment]
> When we think about it, human space isnt that unsafe except for pirate worlds

In the specific context of exponential fleet growth through trading, yes, human space quickly becomes a constant slog through pirate fleets on every jump. Which is by design, to ensure that you can't just keep buying freighters endlessly; you need to branch out into combat capable ships. That's what makes Hai space special; you don't have that constraint.

**Comment by Hurleveur** [comment]
Neither do you have it in Jump reachable areas, meaning they're just better once you can get there.
~~But why reply now~~
Whilst I'm posting I might as well add that I still plan to open a PR to make hai-uhai trade temporarily better, when the devs get to the uhai campaign.
